addons/website_sale_wishlist/models/product_wishlist.py

```diff
@@ -89,4 +49,4 @@ class ProductProduct(models.Model):
 
     def _is_in_wishlist(self):
         self.ensure_one()
-        return self in self.env['product.wishlist'].current().mapped('product_id')
+        return self in self.env['product.wishlist'].current().filtered(lambda wish: wish.displayed_in_cart).mapped('product_id')
```

Comment:
```suggestion
        # displayed_in_cart is used to tell whether a wishlist record is in the actual wishlist or not.
        return self in self.env['product.wishlist'].current().filtered(lambda wish: wish.displayed_in_cart).mapped('product_id')
```
This comment could also be moved on the definition of `displayed_in_cart`